docs/07_code_coverage_artifact_import.md

```diff
@@ -0,0 +1,347 @@
+
+# Motivation
+> What problem is this project solving?
+
+Automatically collect code coverage metrics from various CI providers and build jobs.
+
+# Goals
+> Identify success metrics and measurable goals.
+
+* CI can export coverage metric as an artifact.
+* Exported artifacts are accessible via the CI's API.
+
+# Non-Goals
+> Identify what's not in scope.
+
+Implementing code coverage artifact creation is out of scope because that's job specific.
+
+# Design
+> Explain and diagram the technical design
+
+`CI` → `Artifacts storage` → `API` → `Metrics app`
+
+> Identify risks and edge cases
+
+* The approach is sensitive to breaks/changes of CI artifacts storage and API server - if one changes the importing flow may require changes as well. 
+
+# API
+> What will the proposed API look like?
+
+Add a build step to export coverage metric as an artifact - so, later on, it is accessible via API from artifacts related to the build.
+
+# Bitrise
+
+## Deploy artifacts
+Artifacts are deployed into bitrise.io with the help of the **Deploy to Bitrise.io** Step. Use the following steps to get generated files accessible within build:
+1. Generate artifacts during one of the workflow steps.
+2. Insert the Deploy to Bitrise.io Step AFTER the Step(s) that generate the artifacts.
+3. Change the target directory path in the **Deploy directory or file path** field under the **Config** section of the step if required.
+   For example `$BITRISE_SOURCE_DIR/build/` or `$BITRISE_SOURCE_DIR/coverage/`. 
+
+More information [here](https://devcenter.bitrise.io/builds/build-artifacts-online/).
+
+## Create API Access Token
+The authorization is required to communicate with Bitrise API. For this purpose, the user can generate a Personal access token.
+1. On Bitrise navigate to your [Account settings](https://www.bitrise.io/me/profile) and click on the [Security tab](https://app.bitrise.io/me/profile#/security).
+2. Scroll to the **Personal access token (BETA)** section and press the **Generate new** button.
+3. Provide a token description (must be unique) and select expiration duration in the popup. **Save & Continue** to proceed.
+4. Next popup will contain the token generated. **Make sure to copy your new personal access token. You won’t be able to see it again!**
+5. Use a new token in requests by setting `Authorization: <access token>` header.
+
+For more detailed instructions please see [discussion](https://discuss.bitrise.io/t/personal-access-tokens-beta/1383) and [documentation](https://devcenter.bitrise.io/jp/api/authentication/). 
+
+## Retrieve artifacts via API
+Bitrise provides several endpoints to work with builds and artifacts. Here is a list of endpoints you can use to import code coverage artifact:
+ - **GET** `/apps/{app-slug}/builds` responses with a list of builds. Each build contains the `slug` field which stands for an ID of this build. Select the one you want to load code coverage artifact for and proceed to the next endpoint.
```

Comment:
Should we add more information about "app-slug"? Where to get it (format, etc...) like in CircleCI?